### PR TITLE
Adding nn.ZeroPad1d and nn.ZeroPad3d

### LINKED ISSUE #95796: Can we add nn.ZeroPad1d/3d like nn.ZeroPad2d ?
### 🚀 The feature, motivation and pitch

Feature request: Add nn.ZeroPad1d/3d like nn.ZeroPad2d.
Motivation:
1. Improve usability.
2. To be consistent with the other padding layers. `Reflection`, `Replication` and `Constant` Padding all have 1d/2d/3d available. Only `ZeroPad` is 2d exclusive (for now). We should do this small improvement to make it a little bit nicer.

### Alternatives

_No response_

### Additional context

_No response_

cc @albanD @mruberry @jbschlosser @walterddr @saketh-are

### PR DESCRIPTION
**Pull request by rishub-tamirisa**
Fixes #95796 

### Implementation
Adds python implementation for `nn.ZeroPad1d` and `nn.ZeroPad3d` in `torch/nn/modules/padding.py`.

Adds cpp implementation for `nn::ZeroPad1d` and `nn::ZeroPad3d` in the following 3 files, refactored with templates similarly to `nn::ConstantPad`'s implementation: <br>
- `torch/crsc/api/include/torch/nn/modules/padding.h`
- `torch/csrc/api/include/torch/nn/options/padding.h`
- `torch/csrc/api/src/nn/modules/padding.cpp`

Also added relevant definitions in `torch/nn/modules/__init__.py`.
### Testing
Adds the following tests:
-  cpp tests of similar length and structure as `ConstantPad` and the existing `ZeroPad2d` impl in `test/cpp/api/modules.cpp`
- cpp API parity tests in `torch/testing/_internal/common_nn.py`
- module init tests in `test/test_module_init.py`

Also added relevant definitions in `test/cpp_api_parity/parity-tracker.md`

cc @soulitzer @albanD @mruberry @jbschlosser @walterddr @saketh-are

### CODE REVIEW

File: torch/nn/modules/padding.py [hunk 1]
```diff
@@ -524,3 +576,47 @@ def __init__(self, padding: _size_4_t) -> None:
...
+
+    padding: Tuple[int, int, int, int, int, int]
+
+    def __init__(self, padding: _size_6_t) -> None:
+        super().__init__(padding, 0.)
+
+    def extra_repr(self) -> str:
+        return '{}'.format(self.padding)
+    
+
```
Comment (soulitzer) [inline torch/nn/modules/padding.py:622]: this red symbol means that the lint check for end of file newlines failed
lint expects there to be exactly one new line at the end of each file
you might want to double check the other files modified

**Review by soulitzer — COMMENTED** [review]
One thing you'll want to add is an entry in `docs/source/nn.rst`
For testing, the additions in this PR seem fine, though we tend to use `torch/testing/_internal/common_modules.py` now I think

@jbschlosser do you want to take on the review?

**Comment by rishub-tamirisa** [comment]
@soulitzer Thanks for the quick response! I just updated the newlines and added the corresponding entries in `nn.rst`.

**Comment by rishub-tamirisa** [comment]
@soulitzer Apologies, there was a doctest error in `nn.ZeroPad3d` that was causing several failed tests and the lintrunner error was mistakenly not resolved due to an incorrect indentation on the last line - these should be resolved now.

**Review by soulitzer — APPROVED** [review]
Looks good, thanks!

**Review by jbschlosser — COMMENTED** [review]
Thanks for the hard work @rishub-tamirisa! Looks great :)

> I didn't use common_modules.py because I didn't see a precedent of other padding operations being tested there.

Note that `common_modules.py` is the "new way" to do testing, replacing the old dicts in `common_nn.py`. The lack of padding operations there is just an unfortunate symptom of the fact we don't have full coverage there yet. I won't hold up this PR on that, but if you wanted to add entries to be tested there as well, it would be greatly appreciated :)

**Comment by rishub-tamirisa** [comment]
> Note that `common_modules.py` is the "new way" to do testing, replacing the old dicts in `common_nn.py`. The lack of padding operations there is just an unfortunate symptom of the fact we don't have full coverage there yet. 

@jbschlosser Understood - I'm happy to add entries there. Just to be clear, should I be adding new tests to `common_modules.py` and keeping the current ones in `common_nn.py`, or actually moving what's written in `common_modules.py` over? Additionally, should I open a separate PR for doing this for all padding ops?

**Comment by soulitzer** [comment]
@rishub-tamirisa Feel free to remove the ones in `common_nn.py`, the new test will subsume this one, they should test the same things and more. 

> Additionally, should I open a separate PR for doing this for all padding ops?

This would be great! We'd be happy to review and merge any PRs doing so.

**Comment by rishub-tamirisa** [comment]
> they should test the same things and more.

@soulitzer @jbschlosser Is there another location where cpp API parity tests - that don't rely on the dicts in `common_nn.py` and instead use `module_db` from `common_modules.py` - are being performed? I just want to be sure that I wouldn't be causing those tests for these ops to be lost if I naively remove them from `common_nn.py`. 

Tracing through the code I've assumed `test/cpp_api_parity/module_impl_check.py` is responsible for this check, which seems to only be used with the test params dicts from `common_nn.py` in `test/test_cpp_api_parity.py`.

**Pull request closed without merging**